stable_baselines/deepq/replay_buffer.py

```diff
@@ -63,6 +81,8 @@ def sample(self, batch_size, **_kwargs):
             - done_mask: (numpy bool) done_mask[i] = 1 if executing act_batch[i] resulted in the end of an episode
                 and 0 otherwise.
         """
+        # TODO(araffin): should we ensure sample with no replacement?
```

Comment:
the original [thesis](https://apps.dtic.mil/dtic/tr/fulltext/u2/a261434.pdf) that talks about experience replay, doesn't seem to specify this.

In the original DQN paper's code https://github.com/deepmind/dqn/blob/master/dqn/TransitionTable.lua#L102 they use uniform sampling, so keep uniform as default and maybe add a flag?

Reply:
I think with a buffer size >> batch size, this does not make any difference (which is the case most of the time I would say)

Reply:
not so sure, if you look at the [probability equation](https://en.wikipedia.org/wiki/Birthday_problem#Calculating_the_probability) for `buffer_size = 50000` and `batch_size = 32`, you get p(n) ~ 0.990, so unlikely but still relevent

Reply:
me neither, as the gradient is average over those samples